src/clr-angular/forms/datepicker/date-input.spec.ts

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Component, DebugElement, ViewChild} from "@angular/core";
+import {ComponentFixture, fakeAsync, TestBed, tick} from "@angular/core/testing";
+import {FormsModule} from "@angular/forms";
+import {By} from "@angular/platform-browser";
+
+import {TestContext} from "../../data/datagrid/helpers.spec";
+import {IfOpenService} from "../../utils/conditional/if-open.service";
+import {FormControlService} from "../common/form-control.service";
+import {ClrFormsModule} from "../forms.module";
+
+import {ClrDateContainer} from "./date-container";
+import {ClrDateInput} from "./date-input";
+import {DateIOService} from "./providers/date-io.service";
+import {DateNavigationService} from "./providers/date-navigation.service";
+import {DatepickerEnabledService} from "./providers/datepicker-enabled.service";
+import {MockDatepickerEnabledService} from "./providers/datepicker-enabled.service.mock";
+import {LocaleHelperService} from "./providers/locale-helper.service";
+
+export default function() {
+    describe("Date Input Component", () => {
+        let context: TestContext<ClrDateInput, TestComponent>;
+        let enabledService: MockDatepickerEnabledService;
+        let dateIOService: DateIOService;
+
+        describe("Basics", () => {
+            beforeEach(function() {
+                TestBed.overrideComponent(ClrDateContainer, {
+                    set: {
+                        providers: [
+                            {provide: DatepickerEnabledService, useClass: MockDatepickerEnabledService}, IfOpenService,
+                            DateNavigationService, LocaleHelperService, DateIOService, FormControlService
+                        ]
+                    }
+                });
+
+                context = this.create(ClrDateInput, TestComponent, []);
+
+                enabledService =
+                    <MockDatepickerEnabledService>context.fixture.debugElement.query(By.directive(ClrDateContainer))
+                        .injector.get(DatepickerEnabledService);
+                dateIOService =
+                    context.fixture.debugElement.query(By.directive(ClrDateContainer)).injector.get(DateIOService);
+            });
+
+            describe("Typescript API", () => {
+                it("gets the placeholder from the IO service", () => {
+                    // since we are testing with en-US
+                    expect(context.clarityDirective.placeholderText).toBe("MM/DD/YYYY");
```

Comment:
Test against the value returned by the provider.

Reply:
fixed